distsql/request_builder.go

```diff
@@ -307,8 +308,11 @@ func (builder *RequestBuilder) SetFromInfoSchema(pis interface{}) *RequestBuilde
 
 // SetResourceGroupTagger sets the request resource group tagger.
 func (builder *RequestBuilder) SetResourceGroupTagger(sc *stmtctx.StatementContext) *RequestBuilder {
-	if topsqlstate.TopSQLEnabled() {
-		builder.Request.ResourceGroupTagger = sc.GetResourceGroupTagger()
+	tagger := sc.GetResourceGroupTagger()
+	builder.Request.ResourceGroupTagger = func(req *tikvrpc.Request) {
+		if topsqlstate.TopSQLEnabled() {
+			tagger(req)
```

Comment:
This might miss the case that there are long running TiKV requests?

Reply:
yes. But normally 1 RPC requests shouldn't run too long,

Reply:
After TopSQL is enabled, the subsequent RPC request will contain the tag.